terraform/eval_variable.go

```diff
@@ -80,6 +84,14 @@ func (n *EvalTypeCheckVariable) Eval(ctx EvalContext) (interface{}, error) {
 				return nil, fmt.Errorf("variable %s%s should be type %s, got %T",
 					name, modulePathDescription, declaredType.Printable(), proposedValue)
 			}
```

Comment:
Just caught this error as a user when I didn't declare a `type` on a module variable and then passed in a `map`:

```
❯ tf plan
Error configuring: 1 error(s) occurred:

* variable tags in module mod should be type string, got map[string]interface {}
```

If I weren't a Go developer this would probably scare me. 😄 

Happy to wait for error message twiddling for a subsequent PR though.